sdk/cosmos/azure-cosmos/azure/cosmos/_retry_utility.py

```diff
@@ -198,6 +214,38 @@ def ExecuteFunction(function, *args, **kwargs):
     """
     return function(*args, **kwargs)
 
+def _has_write_retryable_headers(request_headers):
+    if request_headers.get(HttpHeaders.ThinClientProxyResourceType) == "docs":
+        return True
+    return False
+
+def _has_read_retryable_headers(request_headers):
+    if _OperationType.IsReadOnlyOperation(request_headers.get(HttpHeaders.ThinClientProxyOperationType)):
+        return True
+    return False
+
+def _handle_service_request_retries(request, client, response_retry_policy, *args):
```

Comment:
L227 can be `def _handle_service_request_retries(request, client, request_retry_policy, *args):` for readability.